pants

```diff
@@ -361,7 +361,7 @@ function bootstrap_pants {
         # Grab the latest pip, but don't advance setuptools past 58 which drops support for the
         # `setup` kwarg `use_2to3` which Pants 1.x sdist dependencies (pystache) use.
         "${staging_dir}/install/bin/pip" install --quiet -U pip "setuptools<58" && \
-        "${staging_dir}/install/bin/pip" install ${maybe_find_links} --quiet --progress-bar off "${pants_requirement}"
+        "${staging_dir}/install/bin/pip" install ${maybe_find_links} --ignore-installed --quiet --progress-bar off "${pants_requirement}"
```

Comment:
This doesn't make sense even if it works. The venv this is running in was created fresh 2 non-comment lines above and we are not using `--system-site-packages` when creating that fresh venv. As such, there should be nothing installed - except `pip`, `setuptools` and `wheel`.

That gives me pause. Do you understand how this works - aka why pip is seeing anything as installed in this fresh venv at all? Do you have things like PYTHONPATH set in your execution environment that are modifying what might be seen to include things outside this fresh venv perhaps?

At the very least a comment explaining what's going on here if you understand it (I don't). Perhaps also the less dangerous `--upgrade --upgrade-strategy eager` if that works.

Reply:
> Do you have things like PYTHONPATH set in your execution environment 

Yup!

> Perhaps also the less dangerous --upgrade --upgrade-strategy eager if that works.

It has the same issue.

Reply:
There's `--isolated` but that ignores user config which I don't think is strictly-speaking what we want.

Reply:
This is easily reproducible, will add to issue.

Reply:
https://github.com/pantsbuild/setup/issues/120#issuecomment-1087995877

Reply:
OK, thanks for checking. How about amending scrub_PEX_env_vars to also scrub PYTHONPATH?

Reply:
OR, if you stick with the current, please just add a comment saying how the heck there can be anything installed to ignore - because its not installed, but it is on some ambient PYTHONPATH.

Reply:
Scrubbing the env is likely the right approach.
I'll try it out tomorrow.